mockintosh/stats.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+"""
+.. module:: __init__
+    :synopsis: module that contains server classes.
+"""
+
+
+class BaseStats:
+    def __init__(self):
+        self.parent = None
+        self.children = None
+        self.request_counter = 0
+        self.status_code_distribution = {}
+        self.avg_response_time_in_microseconds = 0
+
+    def increase_request_counter(self):
+        self.request_counter += 1
+        if not hasattr(self, 'parent'):
+            return
+
+        method = getattr(self.parent, "increase_request_counter", None)
+        if callable(method):
+            method()
+
+    def add_request_elapsed_time(self, elapsed_time_in_microseconds):
+        self.avg_response_time_in_microseconds = (
```

Comment:
A better alternative is to store _sum_ of all response times and then divide it by number of requests upon writing into JSON. Spends less CPU and suffers less from any kind of rounding problems.

Reply:
Fixed with https://github.com/up9inc/mockintosh/commit/4d27f4bc5f58a1cd0ba182fad16cf8dc8b9a0622